src/runtime/patch/child_process.js

```diff
@@ -0,0 +1,203 @@
+// 1 - first add the arguments starting with -- or -
+// 2 - add the main static fs runtime path
+// 3 - add the static-fs-volumes flag
+// 4 - add the main entry
+// 5 - add the rest of the args
+//
+// However for fork the order is different as fork don't accept
+// node args starting by -- or - unless we set the options.execArgs
+// So for fork (when mainStaticFsRuntimePath === null) the order is
+//
+// 1 - add the static-fs-volumes flag
+// 2 - add the main entry
+// 3 - add the rest of the args
+function buildStaticFsArgs(args, mainStaticFsRuntimePath, staticFsVolumesPaths, mainEntry = null) {
+  const sanitizedArgs = Array.isArray(args) ? args : typeof args === 'string' ? args.split(' ') : [];
+  const builtArgs = [];
+  let toAddMetaArgs = true;
+
+  const iterableArgs = [...sanitizedArgs];
+
+  do {
+    const wasArrayEmptyBeforeArg = iterableArgs.length === 0;
+    const arg = iterableArgs.shift();
+
+    if (!arg && !wasArrayEmptyBeforeArg) {
+      return;
+    }
+
+    if ((typeof arg === 'string' && arg.startsWith('-') && mainStaticFsRuntimePath) || !toAddMetaArgs) {
+      builtArgs.push(arg);
+      continue;
+    }
+
+    toAddMetaArgs = false;
+
+    if (mainStaticFsRuntimePath) {
+      builtArgs.push(mainStaticFsRuntimePath);
+    }
+
+    builtArgs.push(`--static-fs-volumes=${staticFsVolumesPaths.join(',')}`);
+
+    if (mainEntry) {
+      builtArgs.push(mainEntry);
+    }
+
+    builtArgs.push(arg);
+  } while (iterableArgs.length > 0);
+
+  if (!builtArgs.length) {
+    throw new Error('Something went wrong building the static fs args for the child_process functions');
+  }
+
+  return builtArgs;
+}
+
+function indexAfterStartsWith(command, text) {
+  return command.startsWith(text) ? text.length : -1;
+}
+
+function isNode(path) {
+  return possibilities.indexOf(path) > -1;
+}
+
+function startsWithNode(command) {
+  if (command.charAt(0) === '"') {
+    // check includes quotes
+    for (const each of possibilities) {
+      const val = indexAfterStartsWith(command, `"${each}" `);
+      if (val > -1) {
+        return val;
+      }
+    }
+  } else {
+    for (const each of possibilities) {
+      const val = indexAfterStartsWith(command, `${each} `);
+      if (val > -1) {
+        return val;
+      }
+    }
+  }
+  return -1;
+}
+
+export function patchChildProcess(sfsRuntime) {
+  const sfs = sfsRuntime.staticfilesystem;
+  const fork = child_process.fork;
+  const spawn = child_process.spawn;
+  const exec = child_process.exec;
+  const spawnSync = child_process.spawnSync;
+  const execSync = child_process.execSync;
+
+  // hot-patch fork so we can make child processes work too.
+  child_process.fork = (modulePath, args, options) => {
+    const sanitizedOptions = args && typeof args === 'object' && args.constructor === Object ? args : options || {};
+    const sanitizedArgs = Array.isArray(args) ? args : [];
+    const optsEnv = Object.assign(process.env, sanitizedOptions.env || {});
```

Comment:
The Node api for that is here https://nodejs.org/docs/latest-v10.x/api/child_process.html#child_process_child_process_fork_modulepath_args_options

When you fork a process, the child will inherit the env variables from the parent. However the node api allows you to setup other env vars which in case it happens should be provided to the forked child process. 

So, 

- if you didn't set child_process.fork env option -> process.env would be auto provided. 
- if child_process.fork env options are provided they should be provided instead only.

I was failing on the last one. 
I've pushed a fix to solve this. The only thing I'm adding always to the child process are the static-fs bootstrap env vars which are needed in order to get it working :)

Reply:
I noticed this and a few others are marked as resolved. Usually it's best to ensure the original poster is satisfied with the response before resolving otherwise the discussion will get lost. If this was discuss over Zoom/Slack, it would be good to post that here before resolving so there is context.

Reply:
👍